addons/reloadlaunchers/functions/fnc_getLoadableMissiles.sqf

```diff
@@ -1,29 +1,23 @@
 #include "script_component.hpp"
 /*
- * Author: commy2
+ * Author: commy2, johnb43
  * Return all magazine types from reloaders inventory that are compatible with given weapon.
  *
  * Arguments:
- * 0: Unit to to the reload <OBJECT>
- * 1: A launcher <OBJECT>
+ * 0: Unit to execute the reload <OBJECT>
+ * 1: Launcher name <STRING>
  *
  * Return Value:
  * Reloable magazines <ARRAY>
  *
  * Example:
- * [bob, launcher] call ace_reloadlaunchers_fnc_getLoadableMissiles
+ * [player, "launch_RPG32_F"] call ace_reloadlaunchers_fnc_getLoadableMissiles
  *
  * Public: No
  */
 
 params ["_unit", "_weapon"];
 TRACE_2("params",_unit,_weapon);
 
-// get available magazines of reloader, Note: "magazines" does not include currently loaded magazines
-private _magazines = magazines _unit;
-
-// case sensitvity
-_magazines = _magazines apply {toLower _x};
-
-// get reloaders magazine types compatible with targets launcher. No duplicates.
-getArray (configFile >> "CfgWeapons" >> _weapon >> "magazines") select {toLower _x in _magazines} // return
+// Look for primary muzzle magazines only
+(compatibleMagazines [_weapon, "this"]) arrayIntersect (magazines _unit)
```

Comment:
Again, slow, but faster than before I guess.

Reply:
I don't think there is much we can do to speed this up.

Reply:
> I don't think there is much we can do to speed this up.

Is it worth caching?

Reply:
`(compatibleMagazines [secondaryWeapon player, "this"]) arrayIntersect (magazines player);` is ~0.0043ms (with 16 mags)
I don't think it's a problem